doc/reference/cmdline.rst

```diff
@@ -548,4 +548,124 @@ Scaffolding is available via the :command:`odoo-bin scaffold` subcommand.
+
+.. code-block:: console
+
+    $ odoo-bin cloc --addons-path=dirs -d database
+
+In the latter mode, only the custom code is accounted for.
+You can also use a configuration file -c configuration.conf instead of the addons-path
+
+.. code-block:: console
+
+    $ odoo-bin cloc -c configuration.conf -d database
+
+both mode can be used simultaneously to count the line for a given folder
+in addition of the lines in the given database.
+
+.. code-block:: console
+
+    $ odoo-bin cloc --addons-path=dirs -d database -p module_path
+
+
+The verbose option show the details per files.
+
+
+.. option:: --verbose, -v
+
+
+Which files Odoo Cloc counts ?
+------------------------------
+
+Database mode
+'''''''''''''
+
+Odoo Cloc counts the lines in each file of non standard installed modules in a
+given database. In addition, it counts the python line from server action and
+custom computed field stored directly in the database, in other words, not defined
+in any module or defined in an importable module.
+
+Some files are excluded from the count by default
+
+- The manifest (__manifest__.py or __openerp__.py)
+- The content of the folder static/lib
+- The tests defined in folder tests and static/tests
+- The xml files declared in demo or demo_xml section of the manifest
+
+Finally, it's possible to define in each module a list of files to ignore
+with the directive "cloc_exclude" in the manifest.
+
+.. code-block:: python
+
+    "cloc_exclude": [
+        "lib/common.py", # exclude a single file
+        "data/*.xml", # exclude all xml file in a specific folder
+        "example/**/*", # exclude all file in the folder and its subfolder
+    ]
+
+The syntax "\*\*/\*" can be used to ignore a complete module. In that case,
+the module will not be covered by the maintenance service.
+
+For more information about the syntax,
+see `glob <https://docs.python.org/3.6/library/pathlib.html#pathlib.Path.glob>`_
+
+
+
+Path mode
+'''''''''
+
+Odoo Cloc counts as described in database mode if a manifest file is present in
+the given folder. Otherwise, it counts all files.
+
+
+How does cloc make the difference between standard and custom module ?
+----------------------------------------------------------------------
+
+Odoo Cloc uses the following heuristic: If the module is in the same addons path as
+the modules base, web or web_enterprise then the module is considered as standard.
+
+Therefore, we strongly advise to have your custom module in a separate addons_folder
+
+
+Error Handling
+--------------
+
+Some file cannot be counted by Odoo Cloc.
+Those file are reported at the end of the output.
+
+Max file size exceeded
+''''''''''''''''''''''
+
+Odoo Cloc rejects any file bigger than 25MB. Usually source files are smaller
+than 1 MB, if you have that kind of file reported, it's probably
+- a generated xml file that contains lot of data, exclude it in the manifest
+- a javascript library that should be placed in static/lib
```

Comment:
```suggestion
Odoo Cloc rejects any file larger than 25MB. Usually, source files are smaller
than 1 MB. If a file is rejected, it may be:

- A generated xml file that contains lots of data. Exclude it in the manifest.
- A JavaScript library that should be placed in the :file:`static/lib` folder.
```